test/integration/rt_channel_test.exs

```diff
@@ -2129,32 +2109,85 @@ defmodule Realtime.Integration.RtChannelTest do
     end
   end
 
-  defp token_valid(tenant, role, claims \\ %{}), do: generate_token(tenant, Map.put(claims, :role, role))
-  defp token_no_role(tenant), do: generate_token(tenant)
+  test "tracks and untracks properly channels", %{tenant: tenant} do
+    assert [] = Tracker.list_pids()
 
-  defp generate_token(tenant, claims \\ %{}) do
-    claims =
-      Map.merge(
-        %{ref: "127.0.0.1", iat: System.system_time(:second), exp: System.system_time(:second) + 604_800},
-        claims
-      )
+    {socket, _} = get_connection(tenant)
+    config = %{broadcast: %{self: true}, private: false, presence: %{enabled: false}}
+
+    topics =
+      for _ <- 1..10 do
+        topic = "realtime:#{random_string()}"
+        :ok = WebsocketClient.join(socket, topic, %{config: config})
+        assert_receive %Message{topic: ^topic, event: "phx_reply"}, 500
+        topic
+      end
+
+    assert [{_pid, count}] = Tracker.list_pids()
+    assert count == length(topics)
 
-    {:ok, generate_jwt_token(tenant, claims)}
+    for topic <- topics do
+      :ok = WebsocketClient.leave(socket, topic, %{})
+      assert_receive %Message{topic: ^topic, event: "phx_close"}, 500
+    end
+
+    assert [{_pid, 0}] = Tracker.list_pids()
   end
 
-  defp get_connection(
-         tenant,
-         role \\ "anon",
-         claims \\ %{},
-         params \\ %{vsn: "1.0.0", log_level: :warning}
-       ) do
-    params = Enum.reduce(params, "", fn {k, v}, acc -> "#{acc}&#{k}=#{v}" end)
-    uri = "#{uri(tenant)}?#{params}"
+  test "failed connections are present in tracker with counter counter lower than 0 so they are actioned on by tracker",
+       %{tenant: tenant} do
+    assert [] = Tracker.list_pids()
+
+    {socket, _} = get_connection(tenant)
+    config = %{broadcast: %{self: true}, private: true, presence: %{enabled: false}}
 
-    with {:ok, token} <- token_valid(tenant, role, claims),
-         {:ok, socket} <- WebsocketClient.connect(self(), uri, @serializer, [{"x-api-key", token}]) do
-      {socket, token}
+    for _ <- 1..10 do
+      topic = "realtime:#{random_string()}"
+      :ok = WebsocketClient.join(socket, topic, %{config: config})
+      assert_receive %Message{topic: ^topic, event: "phx_reply", payload: %{"status" => "error"}}, 500
     end
+
+    assert [{_pid, count}] = Tracker.list_pids()
+    assert count == 0
+  end
+
+  test "failed connections but one succeeds properly tracks",
```

Comment:
Could we make the test check if the websocket connection is killed? I think if we start a Tracker with a low interval it should work?